pkg/gw/safeset_handler_test.go

```diff
@@ -0,0 +1,125 @@
+package gw
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	immuclient "github.com/codenotary/immudb/pkg/client"
+	immudb "github.com/codenotary/immudb/pkg/server"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+)
+
+func generateRandomTCPPort() int {
+	rand.Seed(time.Now().UnixNano())
+	min := 1024
+	max := 64000
+	return rand.Intn(max - min + 1)
+}
+
+func TestSafeset(t *testing.T) {
+	key := base64.StdEncoding.EncodeToString([]byte("Pablo"))
+	value := base64.StdEncoding.EncodeToString([]byte("Picasso"))
+	tcpPort := generateRandomTCPPort()
+	op := immudb.DefaultOptions().WithPort(tcpPort).WithDir("db_" + strconv.FormatInt(int64(tcpPort), 10))
+	s := immudb.DefaultServer().WithOptions(op)
+	go s.Start()
+	time.Sleep(2 * time.Second)
+	defer s.Stop()
+	defer os.RemoveAll(op.Dir)
+
+	ic, err := immuclient.NewImmuClient(immuclient.DefaultOptions().WithPort(tcpPort))
+	if err != nil {
+		t.Errorf("unable to instantiate client: %s", err)
+		return
+	}
+	mux := runtime.NewServeMux()
+	ssh := NewSafesetHandler(mux, ic)
+
+	tt := []struct {
+		test           string
+		payload        string
+		wantStatus     int
+		wantFieldName  string
+		wantFieldValue interface{}
+	}{
+		{ //Pablo = UGFibG8=
+			//Picasso = UGljYXNzbw==
+			"Send correct request",
+			`{"kv": {"key": "` + key + `", "value": "` + value + `"	} }`,
+			200,
```

Comment:
please use a more meaningful constant like http.StatusOK

Reply:
Actually the status is not used in the tests yet as mentioned in the comment at the end of the function.
#263 is a blocker for that